subprojects/logging/src/test/groovy/org/gradle/internal/logging/console/ProgressOperationTest.groovy

```diff
@@ -43,4 +43,22 @@ class ProgressOperationTest extends Specification {
         expect:
         progressOperation.getMessage() == null
     }
+
+    def "allows children to be managed"() {
+        given:
+        ProgressOperation progressOperation = new ProgressOperation("SHORT_DESCRIPTION", "STATUS", "CATEGORY", new OperationIdentifier(1), null)
+        def mockOperation = Mock(ProgressOperation)
+
```

Comment:
:o: For completeness we should ensure that you can call hasChildren() before any children are added with no NPE